### ISSUE TITLE
.scan methods returns array of undefined

### ISSUE THREAD

**Issue by Alvyre** [glayzzle/php-reflection#20]
Hello!

First of all, thanks for your work and for this project.

I'm encountering some issues about using the method `.scan()` of the workspace. Indeed, When I load a PHP file with `.parse()` method, I got the file object but with `scan` one, it only return an array of undefined values. I check the path of the workspace, I use the same for the workspace, and I checked the folder path too.

Do you have some ideas about why it doesn't work?

Thanks

**Comment by ichiriac** [comment]
The scan method returns a promise that resolves an array of objects. Could you show me a sample of code in order to see how you try to use them.

Note that you can on the promise closure also use meanwhile `repositiory.eachFile((file) => { ... });` it will give you each file structure

**Comment by Alvyre** [comment]
Hello @ichiriac ,

Thanks for your reply, this is a piece of a test class I've made in nodejs : 
Please feel free to ask any question about the code and sorry about dirty code incoming.

```js
'use strict';

// import fs from 'fs';
import _ from 'lodash';
import config from './config.json';
import reflection from 'php-reflection';

class JSParser {
    constructor(props) {
        this.workspacePath = props.workspace;
        this.workspace = new reflection.Repository(this.workspacePath, {
            exclude: ['.git', '.svn'],
            include: ['./'],
            ext: ['*.php'],
            scanVars: true,
            scanExpr: true,
            encoding: 'utf8'
        });
        // this.bundles = [];

        this._loadBundles();
    }

    async _loadBundles() {
        const bundles = Object.values(config.bundles);
        const result = this.workspace.scan(bundles);
        result.then((item) => {
            console.log(item.getFile())
        }).catch(err => {
            console.log(err)
        })
        
        console.log(this.bundles);
    }



};

export default new JSParser({workspace: '<path_to_php_files>'});
```

config.json file:
```json
{
    "bundles": {
        "Bundle1": "<path_to_bundle>",
        "Bundle2": "<path_to_bundle>",
        "Bundle3": "<path_to_bundle>",
        "Bundle4": "<path_to_bundle>",
        "Bundle5": "<path_to_bundle>"
    }
}
```

**Comment by ichiriac** [comment]
Take a look at the API of the workspace class : https://github.com/glayzzle/php-reflection/blob/master/docs/src/REPOSITORY.md

And here the API of the file node : 
https://github.com/glayzzle/php-reflection/blob/master/docs/src/FILE.md

**Comment by Alvyre** [comment]
Okay I see, thank you for your explanations! I will try to dig this way! :+1:

**Comment by Alvyre** [comment]
Hello again!

I come with new issues, when I try to use your code, I got the following error : 

```bash
TypeError: this.db.readIndex is not a function
    at repository.getByType (/home/alvyre/Projects/testParser/node_modules/php-reflection/src/repository.js:147:13)
    at result.then (/home/alvyre/Projects/testParser/dist/utils/JSParser/index.js:37:22)
    at <anonymous>
```

is it related to my node version ? (8.12.0)

**Comment by ichiriac** [comment]
It seems you have an old release, actually the code is using the filter function :
https://github.com/glayzzle/php-reflection/blob/master/src/repository.js#L147

Try to install it from master : npm install --save https://github.com/glayzzle/php-reflection

**Comment by Alvyre** [comment]
Hello, so I got a new error with the method `each`
I logged the repository and the error if it can help you:
```cmd
repository {
  options: 
   { exclude: [ '.git', '.svn' ],
     include: [ './' ],
     ext: [ '*.php' ],
     scanVars: true,
     scanExpr: true,
     scanDocs: true,
     encoding: 'utf8',
     forkWorker: false,
     cacheByFileDate: true,
     cacheByFileSize: true,
     cacheByFileHash: true,
     lazyCache: false,
     debug: false,
     shardSize: 256,
     shardCapacity: 512 },
  db: 
   Graph {
     _hash: 256,
     _capacity: 512,
     _nextId: 10311,
     _shards: 
      [ [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object] ],
     _indexes: 
      [ [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        <1 empty item>,
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        <1 empty item>,
        [Object],
        [Object],
        [Object],
        <1 empty item>,
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        <1 empty item>,
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        <1 empty item>,
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        <1 empty item>,
        [Object],
        [Object],
        [Object],
        ... 156 more items ],
     repository: [Circular] },
  _regex: [ /^.*\.php$/ ],
  _pending: {},
  counter: 
   { total: 175,
     loading: 0,
     loaded: 175,
     error: 0,
     symbols: 10136,
     size: 1366241 },
  directory: '/home/alvyre/projects/parser',
  domain: null,
  _events: {},
  _eventsCount: 0,
  _maxListeners: undefined }
```
```bash
TypeError: Cannot read property 'length' of undefined
    at node_modules/php-reflection/src/repository.js:148:19
    at node_modules/grafine/src/index.js:158:30
    at Map.forEach (<anonymous>)
    at Index.filter (node_modules/grafine/src/index.js:157:18)
    at grafine.graph.Graph.filter (node_modules/grafine/src/graph.js:234:39)
    at repository.getByType (node_modules/php-reflection/src/repository.js:147:13)
```

**Comment by ichiriac** [comment]
bugfix released in 2.3.1
